### ISSUE TITLE
Cursor Flickering when used with Nvim, but NOT vim

### ISSUE THREAD

**Issue by DanielBarton446** [neoclide/coc.nvim#2973]
## versions

vim version: NVIM v0.4.3
node version: v12.20.1
coc.nvim version: 0.0.80-98a0c6db19
coc.nvim directory: /home/daniel/.vim/plugged/coc.nvim
term: xterm-256color
platform: linux

## Log of coc.nvim

2021-03-15T23:00:43.010 INFO (pid:5612) [coc-java] - Using java from /usr/lib/jvm/java-11-openjdk-amd64, version: 11
2021-03-15T23:00:43.036 INFO (pid:5612) [plugin] - coc.nvim 0.0.80-98a0c6db19 initialized with node: v12.20.1 after 186>2021-03-15T23:00:43.054 INFO (pid:5612) [coc-java] - JDT Language Server starting at /home/daniel/dev/javaDev/FactorySi>2021-03-15T23:00:43.055 INFO (pid:5612) [services] - registered service "java"
2021-03-15T23:00:43.066 INFO (pid:5612) [language-client-index] - Language server "java" started with 5639
2021-03-15T23:00:53.189 INFO (pid:5612) [completion-complete] - Results from: around
2021-03-15T23:01:09.014 INFO (pid:5612) [attach] - receive notification: highlight []
2021-03-15T23:02:59.646 INFO (pid:5612) [attach] - receive notification: showInfo []

## Primary issue

Whenever I edit a file which has COC Language support, i.e Java and C, the cursor flickers as though it is not being flushed. This is especially apparent when first loading up a Language supported file and COC is initializing. Additional times this behavior is noted is whenever typing in LS supported syntax where completion is offered, the cursor tends to flicker. 

[Specific example using Java LS, though the behavior is general across LS's]
To be more precise as to some visual feedback I am getting, with having Airline status bar set up, when in Insert mode in nvim, after typing one of these LS words, I note that my status bar gets a "0%" -> "50%" -> "0%" and then eventually fully "loads JDT.LS" (this is what I believe this information on the status bar is reflecting). I believe this may have something to do with the flickering, although I don't really have a bead on what is causing this. 

The Odd thing about this is that I don't get this flickering when I edit a file using vim, but I do when I use nvim. Obviously, this suggests this has something to do with nvim. I just transitioned into nvim, and so my nvim.init is literally the three lines that they suggest when moving to nvim through the 

:help nvim-from-vim

command

Note: I am running ubuntu through WSL 1, which god knows what issues might be arising because of this, doubtful to be applicable in this case(I think), but I thought it might be worth noting.

If anyone has a suggestion, please let me know.

**Comment by chemzqm** [comment]
What do you means by Cursor Flickering ？

**Comment by DanielBarton446** [comment]
Apologies for the lack of precision. What I mean by this is is that the cursor appears to be "blinking" at random intervals. I have attached a mp4 with a short, less pronounced version of my experiences.
Note: When I attempt to convert to GIF, the GIF process removes several frames where the flickering shows, hence the mp4 upload.

https://user-images.githubusercontent.com/47621061/111271134-c9471b80-85fe-11eb-9afd-c8f0754dab77.mp4

Does this clarify? I don't want to say that it is a "blinking" problem such as 

:set guicursor=a:blinkon100

or anything of this nature, in fact I changed this setting to have the cursor NOT blink with the option:

:set guicursor=a:blinkon0

and yet it gets this "stuttering" or "flickering" -- if there is a technical term for this that I am unaware of, please let me know.

I hope this gets to the point

**Comment by chemzqm** [comment]
No idea, tried NVIM v0.4.3 with airline.vim and coc.nvim on ubuntu.
Might be bug of neovim with your terminal, upgrade neovim might help.

**Comment by DanielBarton446** [comment]
Turns out it seems to have fixed itself. I think it was an issue with my shell potentially, ZSH, because I updated that as well as a restart of my computer and that seemed to fix the issue. It could very well also be an issue with the terminal I am using, because I am using the  *new* windows terminal, and who knows if that runs into flushing issues ever. None-the-less, I am glad that it is not longer an issue, and that it is not very reproducible. Very nice to see that because that means it's probably not an actual bug.
